doc/sphinx-guides/source/user/dataset-management.rst

```diff
@@ -179,6 +179,43 @@ Additional download options available for tabular data (found in the same drop-d
+
+The following are general guidelines applicable to all programming languages.
+
+- Create a README text file in the top-level directory to introduce your project. It should answer questions that reviewers or reusers would likely have, such as how to install and use your code. 
+- If possible, use free and open-source file formats and software to make your research outputs more reusable and accessible.
+- Make sure your code uses relative file paths instead of absolute (or full) file paths, as they can cause an execution error.
+- Consider testing your code in a clean environment before sharing it, as it could help you identify missing files or dependencies.
+- Consider providing notes (in the README) on the expected code outputs or adding tests in the code, which would ensure that its functionality is intact.
+
+General Guidelines
+******************
+
+Capturing code dependencies will help other researchers recreate the necessary runtime environment. Without it, your code will not be able to run correctly (or at all). 
+
+Many programing languages enable an automatic dependency capture. Here are a few examples:
+
+- If you are using the conda package manager, you can export your environment with the command ``conda env export > environment.yml``. For more information, see the `official documentation <https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#sharing-an-environment>`_.
+- Python has multiple conventions for capturing its dependencies, but probably the best-known one is with the ``requirements.txt`` file, which is created using the command ``pip freeze > requirements. txt``. Managing environments with ``pip`` is explained in the `official documentation <https://docs.python.org/3/tutorial/venv.html#managing-packages-with-pip>`_.
+- If you are using the R programming language, create a file called ``install.R``, and list all library dependencies that your code requires. This file should be executable in R to set up the environment. See also other strategies for capturing the environment proposed by RStudio in the `official documentation <https://environments.rstudio.com>`_.
+- In case you are using multiple programming languages or different versions of the same language, it would be best to use a containerization technology such as Docker. Create a Dockerfile that builds your environment, making sure to specify dependency versions, and deposit it with the rest of the files. Have a look at the `official Docker guide <https://docs.docker.com/language/python/build-images/>`_ for more information.
```

Comment:
While this is possible, writing a sophisticated `Dockerfile` isn't that easy. It might be fine for small use cases, but often these containers start to be part of a pipeline. Those need careful tweaking to not slow down everything and waste computing resources. Maybe we can add a note here about this somehow.

Reply:
Good point! Bad Dockerfiles defeat the purpose of having them in the first place. I'll add info on that also. 

Thanks very much for the comments!!